### PR TITLE
Added canonical url attribute to category attributes

### PR DESCRIPTION
**Pull request by kandrejevs**

### CODE REVIEW

**Comment by alfredsgenkins** [comment]
Is that really it? If yes, then I merge :)

**Comment by kandrejevs** [comment]
@alfredsgenkins 
It seems so, I was copying the same logic Magento uses in their block to inject canonical url in header for category: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Catalog/Block/Category/View.php#L78

As far as I know, this should be enough. There is more complex logic related to saving and generating URLs, but for retrieval this seems the correct solution.

**Comment by eli-l** [comment]
Current implementation produces such url: "https://scandipwa.local/index.php/men"
"index.php" is obsolete and is not allowed in case of ScandiPWA, as it won't be handled correctly by React router, since we are matching React / Magento routers.

**Comment by kandrejevs** [comment]
@eli-l you need to enable server url rewrites in settings, then index.php is removed from urls

**Comment by eli-l** [comment]
@kandrejevs Nice, didn't notice it. Looks fragile, but we shall solve this globally. One small adjustment and LGTM

File: src/Model/Resolver/Category/CanonicalUrl.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        array $args = null
+    )
+    {
+        if (!isset($value['model'])) {
+            throw new LocalizedException(__('"model" value should be specified'));
+        }
+
+        /* @var $category Category */
+        $category = $value['model'];
+        $url = $category->getUrl();
```
Comment (eli-l) [inline src/Model/Resolver/Category/CanonicalUrl.php]: Let's return immediately, avoiding redundant $url

**Comment by eli-l** [comment]
Thank you @kandrejevs

**Pull request merged**